### PR TITLE
AirTimer: Added an Air Timer Widget.

### PR DESCRIPTION
**Pull request by PawBud**
Related to the [wishlisted feature](https://bugs.launchpad.net/mixxx/+bug/1824655) 
This PR aims to add an Air Timer Widget with the Time format of 'h:mm:ss'

### CODE REVIEW

File: src/widget/wairtime.h [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+#include <QTimer>
+
+#include "widget/wlabel.h"
+#include "skin/skincontext.h"
+
+class WAirTime: public WLabel {
+    Q_OBJECT
+  public:
+    explicit WTime(QWidget *parent=nullptr);
```
Comment (PawBud) [inline src/widget/wairtime.h:16]: Must be exchanged with `WAirTimer`

File: src/widget/wairtime.h [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    QString m_sTimeFormat;
+    // m_iInterval defines how often the time will be updated
+    short m_iInterval;
+    // m_iInterval is set to s_iSecondInterval if seconds are shown
+    // otherwise, m_iInterval = s_iMinuteInterval
+    static const short s_iSecondInterval = 100;
+    static const short s_iMinuteInterval = 1000;
+};
+
+#endif /* WTIME_H */
```
Comment (PawBud) [inline src/widget/wairtime.h:37]: Should be WAIRTIME_H instead of WTIME.H

**Comment by PawBud** [comment]
not sure why the pre-commit hook(Code Style issues), did not detect the code style error when I was pushing my commit to the origin. :/

**Comment by ronso0** [comment]
Thanks for starting this!
Instead of the clock widget you may look how the duration is measured in `/recording/recordingmanager` and how `wrecordingduration`(in the tool bar) and `/library/dlgrecording`(library feature) are connected.

For starting the timer you probably need to connect to [BroadcastManager::slotControlEnabled](https://github.com/mixxxdj/mixxx/blob/main/src/broadcast/broadcastmanager.cpp#L90)

**Comment by Be-ing** [comment]
As I said on Zulip, I think we should pause to consider the UX in combination with recording before merging this. I think it would be a good idea to go in steps, first getting this new widget working then getting it to play nice with the recording timer.

**Comment by PawBud** [comment]
Update: It has been decided that we are going to put three separate stop/start/reset buttons for the UI which will help user manipulate the timer manually.

**Comment by PawBud** [comment]
For reference to why this PR is being closed, Kindly read this https://mixxx.zulipchat.com/#narrow/stream/109171-development/topic/Adding.20an.20On-Air.20Timer/near/219724417.

**Pull request closed without merging**